doc/specifications/proposed/facets.md

```diff
@@ -0,0 +1,303 @@
+with an additional filter (set of filters). Together with the global
+flag this can be used to list facet values in multi-select fields, but
+we probably should not expose this ElasticSearch feature through our API
+since it seems specific and has no direct counterpart in Solr (as far as
+we know).
+
+The `$filter` property will be used for the facet criterion.
+
+From the documentation this is a a BC break. Since the `$filter`
+property is yet unused this should be OK. We still want to use the
+`$filter` property since the property with the same purpose is also
+called `$filter` on the `Query` class. Those filters will be merged and
+should thus be called the same.
+
+### `FacetBuilder->$global`
+
+Omit (deprecate) this property. The idea behind this was to get the
+facet values across the full data and not get the facet results for the
+current search. As far as I can see there is no such option in Solr.
+Thus this would require an extra query. If there is no strong user
+demand for this we should just omit the parameter. This is supported by
+[ElasticSearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-aggregations-bucket-global-aggregation.html)
+but not by other search backends.
+
+If a user still wants this result they can always run a dedicated search
+without any criteria and the seeked "global" facets. Thus there is a
+migration path for a yet non working feature requiring no additional
+implementation by us.
+
+### Multi-Select Facets
+
+In some use cases the following requirement occurs:
+
+> The user wants to query the content. The result shall contain a facet
+> across some property, for example the publishing date. The user
+> selects a first facet criterion (one day) and then wants to select an
+> additional facet criterion (another day).
+
+With the current implementation selecting the first value will only show
+content following the first selected facet criterion. Thus facet value
+mutli select does not work.
+
+We could add a property to the facet builder which enables retrieval of
+all facet values (limited by the common query / filter) even a facet
+value is already selected. This can be implemented against ElasticSearch
+using a global aggregation with a aggregation filter containing only the
+query filter and **not** the facet filter itself while the content
+filter will still be build from the query filter **and** the facet
+filter.
+
+This can be implmented against other search backends, too (maybe using
+additional queries) while this might be harder for the `$global` and
+`$filter` properties already existing in the `FacetBuilder`.
+
+### Facet Value Storage
+
+We should only store the actual IDs in the search backends, not the
+identifier. If a criterion is missing to query that ID we should just
+implement this (`SectionIdCriterion`). The queries are supposed to be
+more performant this way and it spares us re-indexing of the search
+index if identifiers change.
```

Comment:
Note: Updated api doc to relfect in https://github.com/ezsystems/ezpublish-kernel/commit/13b710d9550677d7b803b4702bd9847c9879a0e7 _(v1.10, implementation in Solr was already doing this)_